### DISCUSSION TITLE
How to add order to postgresql index

### DISCUSSION THREAD

**Discussion by HoneyryderChuck** [jeremyevans/sequel#1913] [General]
While adding a partial index to a postgresql table, I couldn't find any docs on how to setup the order of it.

```ruby
# I was expecting this to work, but it didn't:
index :executed_at, order: :desc
# CREATE INDEX "outbox_run_at_index" ON "outbox" ("run_at")
```

**Comment by jeremyevans** [comment]
You can pass a `Sequel::SQL::OrderedExpression` as the value: `index Sequel.desc(:executed_at)`.  An `:order` option wouldn't work for a composite index where not all expressions are ordered the same way.

**Comment by HoneyryderChuck** [comment]
It did the trick, thx for the support!

Out of curiosity, how would it work if I had the order by two different columns, assuming one "desc" and another "asc"?

**Reply by jeremyevans** [reply]
`index [Sequel.desc(:one), :two]`.  You could use `Sequel.asc(:two)`, but ascending is the default, so there is no need.
